### PR TITLE
Add session Api methods to match server and add global settings service

### PR DESCRIPTION
**Pull request by ReissJarvis**
Add global settings service and match session service methods.

the server will need to be updated to match the call of the delete method for the service.  due to the session resource address mismatching a restful delete address. 

Instead of using opaque tokens, I've used a global setting service. The reason behind this is so we can mock this service when we add tests. Allowing us to create integration tests for the client.
Linked to : https://github.com/tsangste/ChaosSpankr.Home/issues/10

### CODE REVIEW

File: src/app/session.service.ts [hunk 1]
```diff
@@ -1,15 +1,35 @@
...
+  create(){
+    return this.http.post(`${this.config.apiUrl}/sessions`, {});
+  }
+
+  get(sessionId: string): Observable<any> {
+    return this.http.get(`${this.config.apiUrl}?sessionId=${sessionId}`);
+  }
+
+  end(sessionId: string){
+    return this.http.delete(`${this.config.apiUrl}/sessions/${sessionId}`);
```
Comment (ReissJarvis) [inline src/app/session.service.ts]: The url here mismatches the servers url.

server is expecting : "sessions" with body { sessionId )

where it should be : "sessions/:sessionId"

**Comment by tsangste** [comment]
Link with an issue please!

File: src/app/app.module.ts [hunk 2]
```diff
@@ -17,7 +19,7 @@ import { HTTP_PROVIDERS } from '@angular/http';
     RouterModule.forRoot(CardAppRoutes)
   ],
   bootstrap: [AppComponent],
-  providers: [HTTP_PROVIDERS, SessionService]
+  providers: [HTTP_PROVIDERS, GlobalSettingsService, SessionService]
```
Comment (tsangste) [inline src/app/app.module.ts]: remove HTTP_PROIVERS and switch to HttpModule

File: src/app/session.service.ts [hunk 3]
```diff
@@ -1,15 +1,35 @@
 import { Injectable } from '@angular/core';
 import { Observable } from 'rxjs';
 import { Http } from '@angular/http';
+import { GlobalSettingsService } from './globalSettings.service';
 
 @Injectable()
 export class SessionService {
+  config;
```
Comment (tsangste) [inline src/app/session.service.ts]: space between constructor and property.  also I think a space after the class declaration and property.

File: src/app/globalSettings.service.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import { Injectable } from '@angular/core';
+import { Config } from './app.config';
+
+@Injectable()
+export class GlobalSettingsService {
+  config = Config;
```
Comment (tsangste) [inline src/app/globalSettings.service.ts]: space between class and property

File: src/app/globalSettings.service.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+  constructor() {
+  }
+
+  get() {
+    return this.config;
+  }
```
Comment (tsangste) [inline src/app/globalSettings.service.ts]: space between the ending function } and the class ending }

File: src/app/session.service.ts [hunk 6]
```diff
@@ -1,15 +1,35 @@
...
+
+  create(): Observable<any> {
+    return this.http.post(`${this.config.apiUrl}/sessions`, {});
+  }
+
+  get(sessionId: string): Observable<any> {
+    return this.http.get(`${this.config.apiUrl}?sessionId=${sessionId}`);
+  }
+
+  end(sessionId: string): Observable<any> {
```
Comment (quickskape) [inline src/app/session.service.ts]: Should this observable return void? Calling delete on the API has no result?

File: src/app/session.service.ts [hunk 7]
```diff
@@ -1,15 +1,35 @@
...
+import { GlobalSettingsService } from './globalSettings.service';
 
 @Injectable()
 export class SessionService {
+  config;
+  constructor(private http: Http, private globalSettingsService: GlobalSettingsService ) {
+    this.config = globalSettingsService.get();
+  }
+
+  create(): Observable<any> {
```
Comment (tsangste) [inline src/app/session.service.ts]: change Observable<any> to Observable<Response> and could you do it to the rest please

Comment (quickskape) [inline src/app/session.service.ts]: I think it should use the map operator and encapsulate the http types within this service.
Do you agree?

File: src/app/session.service.ts (hunk 6 above)
Comment (ReissJarvis) [inline src/app/session.service.ts]: The delete call should still return a http status code. We can use this to then handle errors

Ill link a RFC standard for what a method should return in slack

Comment (tsangste) [inline src/app/session.service.ts]: should probably return a bool if it was successful or not :p

File: src/app/session.service.ts (hunk 7 above)
Comment (tsangste) [inline src/app/session.service.ts]: I agree with @quickskape but I think it should go in another PR

File: src/app/session.service.ts (hunk 6 above)
Comment (quickskape) [inline src/app/session.service.ts]: Agreed 👍 I still think that we should deal with http behaviour inside this service and not expose it to anything calling this service.
What do you think?

File: src/app/session.service.ts (hunk 7 above)
Comment (quickskape) [inline src/app/session.service.ts]: I would be happy for this to be in another PR. I think it will be more of a concern once we start testing the system.

Should we raise an issue or just put in another PR?

File: src/app/app.config.ts [hunk 8]
```diff
@@ -0,0 +1,3 @@
+export const Config = {
```
Comment (tsangste) [inline src/app/app.config.ts]: should we strong type this?

File: src/app/session.service.ts (hunk 7 above)
Comment (ReissJarvis) [inline src/app/session.service.ts]: sorry im lost, are we saying we subscribe in here and map out to a return type?

File: src/app/app.config.ts (hunk 8 above)
Comment (ReissJarvis) [inline src/app/app.config.ts]: i was thinking this. i could create an interface so when its used, the type should bubble up

Comment (tsangste) [inline src/app/app.config.ts]: agreed 👍

File: src/app/session.service.ts (hunk 7 above)
Comment (quickskape) [inline src/app/session.service.ts]: We wouldn't need to subscribe would we? We could just use the map operator to project out the type removing the unneeded http types

Comment (tsangste) [inline src/app/session.service.ts]: @quickskape raise as a new issue with testing for client stuff and we can refactor this service to return an observable of a friendlier type?

File: src/app/globalSettings.service.ts [hunk 9]
```diff
@@ -0,0 +1,16 @@
+import { Injectable } from '@angular/core';
+import { Config } from './app.config';
+
+@Injectable()
+export class GlobalSettingsService {
+
+  config = Config;
```
Comment (tsangste) [inline src/app/globalSettings.service.ts]: missing IConfig type on here

File: src/app/session.service.ts [hunk 10]
```diff
@@ -1,15 +1,37 @@
 import { Injectable } from '@angular/core';
 import { Observable } from 'rxjs';
 import { Http } from '@angular/http';
+import { GlobalSettingsService } from './globalSettings.service';
 
 @Injectable()
 export class SessionService {
 
-  constructor(private http: Http) {}
+  config;
```
Comment (tsangste) [inline src/app/session.service.ts]: missing IConfig type here

Comment (ReissJarvis) [inline src/app/session.service.ts]: we shouldnt need it. this should bubble up from the intial type

Comment (quickskape) [inline src/app/session.service.ts]: Shall we chat at the next code club about this? Might be one for more of an in depth discussion.

Comment (ReissJarvis) [inline src/app/session.service.ts]: yea, i'm happy with that!

File: src/app/session.service.ts [hunk 11]
```diff
@@ -1,15 +1,38 @@
...
+    return this.http.delete(`${this.config.apiUrl}/sessions/${sessionId}`);
+  }
+
+  setStatus(sessionId: string, state: string): Observable<any> {
+    return this.http.put(`${this.config.apiUrl}/sessions/${sessionId}`, {'state': state});
+  }
+
+  addUser(sessionId: string, userId: string): Observable<any> {
+    return this.http.put(`${this.config.apiUrl}/sessions/${sessionId}/user`, {
+      'userId': userId
```
Comment (tsangste) [inline src/app/session.service.ts]: should this be consistent with the one above or should the one above be consistent with this?

File: src/app/session.service.ts (hunk 6 above)
Comment (ReissJarvis) [inline src/app/session.service.ts]: yea agree with this

File: src/app/session.service.ts (hunk 11 above)
Comment (ReissJarvis) [inline src/app/session.service.ts]: ill keep it consistent with a new line. should be easier to read when looking though this then
